testing/Google.Api.Gax.Testing/FakeApiCall.cs

```diff
@@ -0,0 +1,75 @@
+    /// to be created without gRPC 
+    /// </summary>
+    public static class FakeApiCall
+    {
+        /// <summary>
+        /// Creates an API call from the given async/sync functions.
+        /// </summary>
+        /// <typeparam name="TRequest">Request type</typeparam>
+        /// <typeparam name="TResponse">Response type</typeparam>
+        /// <param name="asyncCall">Function to call for async requests</param>
+        /// <param name="syncCall">Function to call for sync requests</param>
+        /// <param name="baseCallSettings">The default call settings. May be null.</param>
+        /// <param name="clock">The clock to use; defaults to a new <see cref="FakeClock"/>.</param>
+        /// <returns>The new API call.</returns>
+        public static ApiCall<TRequest, TResponse> Create<TRequest, TResponse>(
+            Func<TRequest, CallOptions, Task<TResponse>> asyncCall,
+            Func<TRequest, CallOptions, TResponse> syncCall,
+            CallSettings baseCallSettings = null,
+            IClock clock = null)
+            where TRequest : class, IMessage<TRequest>
+            where TResponse : class, IMessage<TResponse>
+        {
+            clock = clock ?? new FakeClock();
+            baseCallSettings = baseCallSettings ?? new CallSettings();
+            return new ApiCall<TRequest, TResponse>(
+                asyncCall.MapArg((CallSettings cs) => cs.ToCallOptions(clock)),
+                syncCall.MapArg((CallSettings cs) => cs.ToCallOptions(clock)),
+                baseCallSettings);
+        }
+
+        /// <summary>
+        /// Creates an API call from a synchronous function, which is just called synchronously for
+        /// asynchronous requests.
+        /// </summary>
+        /// <remarks>
+        /// For asynchronous requests, the synchronous function is called but within an async
+        /// context; exceptions will still be reported back via the returned task, but the call will
+        /// complete synchronously.
+        /// </remarks>
+        /// <typeparam name="TRequest">Request type</typeparam>
+        /// <typeparam name="TResponse">Response type</typeparam>
+        /// <param name="syncCall">Function to call for sync requests</param>
+        /// <param name="baseCallSettings">The default call settings. May be null.</param>
+        /// <param name="clock">The clock to use; defaults to a new <see cref="FakeClock"/>.</param>
+        /// <returns>The new API call.</returns>
+        public static ApiCall<TRequest, TResponse> Create<TRequest, TResponse>(
+            Func<TRequest, CallOptions, TResponse> syncCall,
+            CallSettings baseCallSettings = null,
+            IClock clock = null)
+            where TRequest : class, IMessage<TRequest>
+            where TResponse : class, IMessage<TResponse> =>
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+            Create(async (request, response) => syncCall(request, response),
```

Comment:
That was my first thought - but that would throw exceptions synchronously. This way, if the syncCall throws, you end up with a faulted task, which is more realistic.